src/scripting/lua_menu_item.cpp

```diff
@@ -101,6 +140,29 @@ static int impl_menuitem_collect(lua_State* L)
         return 0;
 }
 
+static int impl_menuitem_dir(lua_State* L)
+{
+    std::vector<std::string> keys;
+
+	gui2::menu_item& item = luaW_checkmenuitem(L, 1);
+
+	if(dynamic_cast<gui2::multimenu_button*>(item.get_parent())) {
+		keys.push_back("checkbox");
+	} else if(dynamic_cast<gui2::menu_button*>(item.get_parent())) {
+		keys.push_back("icon");
+	} else {
+		keys.push_back("checkbox");
+		keys.push_back("icon");
+		keys.push_back("image");
```

Comment:
I think multimenu_button and menu_button actually _can_ use image – checkbox and icon are certainly restricted as they both occupy column 1, but image goes in column 2. I know of no instances where they use image, but I don't think it needs to be restricted.

Reply:
menu_button can't atm, as it's value is the label.  But once I implement the 'value' attribute (to address the translation thing), I suppose it could.

MMB can't atm, because it's button says something like "Alice, Bob, and 2 others".  Same issue as above if an image replaced label.

Reply:
Are you stating that on the basis that you've _actually tried it_ and it didn't work? Because the entire point of the image is that it replaces the label with an image, so I really think it should already work in both menu_button and multimenu_button. There might be some logic I'm unaware of that makes it not so, though…

Reply:
No, haven't tried it yet.  I think it will work, sort of, as long as you do supply the label and the image/label make sense together.  I'll look into it when I implement the canonical value.